File: Python/reverseLinkedList.py
from typing import Optional

class ListNode:
     def __init__(self, val=0, next=None):
         self.val = val
         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        if head is None:
            return None
        
        if head.next is None:
            return head
        
        temp= head
        prev= None
        after= temp.next

        while (temp!=None):
            temp.next= prev
            prev= temp
            temp= after
            
            if after!=None:
                after= after.next
        
        return prev

File: Java/reverseLinkedList.java
class ListNode{
    
    int val;
    ListNode next;

    public ListNode() {
    }

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
    public ListNode(int val){
        this.val= val;
    }
}

class Solution {
    public ListNode reverseList(ListNode head) {

        if(head==null || head.next==null)
            return head;

        ListNode temp= head;
        ListNode prev= null;
        ListNode next= temp.next;

        while(temp!=null){
            temp.next= prev;
            prev= temp;
            temp= next;

            if(next!=null)
            next= next.next;
        }

        return prev;
    }
}